src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -114,59 +148,143 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
+                Dictionary<ImmutableArray<CharacterSetModificationRule>, ImmutableArray<string>> commitCharacterRulesCache,
+                CompletionService completionService,
+                string? clientName,
+                bool returnTextEdits,
+                SourceText? documentText,
+                TextSpan? defaultSpan,
+                LSP.Range? defaultRange,
+                CancellationToken cancellationToken) where TCompletionItem : LSP.CompletionItem, new()
             {
                 var completeDisplayText = item.DisplayTextPrefix + item.DisplayText + item.DisplayTextSuffix;
+
+                // The TextEdits for override and partial method completions are provided in the resolve handler.
+                // We do not provide them in this handler.
+                item.Properties.TryGetValue("Modifiers", out var modifier);
+                var isOverrideOrPartialMethodCompletion = modifier != null && (modifier.Contains("Override") || modifier.Contains("Partial"));
+
+                LSP.TextEdit? textEdit = null;
+                if (returnTextEdits && !isOverrideOrPartialMethodCompletion)
+                {
+                    textEdit = await GenerateTextEdit(
+                        document, item, completionService, documentText, defaultSpan, defaultRange, cancellationToken).ConfigureAwait(false);
+                }
+
                 var completionItem = new TCompletionItem
                 {
                     Label = completeDisplayText,
-                    InsertText = item.Properties.ContainsKey("InsertionText") ? item.Properties["InsertionText"] : completeDisplayText,
                     SortText = item.SortText,
                     FilterText = item.FilterText,
                     Kind = GetCompletionKind(item.Tags),
                     Data = new CompletionResolveData
                     {
                         TextDocument = request.TextDocument,
                         Position = request.Position,
-                        DisplayText = item.DisplayText,
+                        CompleteDisplayText = completeDisplayText,
                         CompletionTrigger = completionTrigger,
                         ResultId = resultId,
                     },
                     Preselect = item.Rules.SelectionBehavior == CompletionItemSelectionBehavior.HardSelection,
                 };
+
+                // Override and partial method completions are always populated in the resolve handler, so we
+                // leave both TextEdit and InsertText unpopulated in these cases.
+                if (isOverrideOrPartialMethodCompletion)
+                {
+                    // Razor C# is currently the only language client that supports LSP.InsertTextFormat.Snippet.
+                    if (clientName == ProtocolConstants.RazorCSharp)
```

Comment:
Techhhnically C# could utilize snippets too (I added them to the lis); however, it still wouldn't work in Liveshare scenarios because I haven't done the needful there so still probably good to restrict to Razor